posts/2024-09-05-Rust-1.81.0.md

```diff
@@ -0,0 +1,166 @@
+- [`IoSliceMut::advance`](https://doc.rust-lang.org/stable/std/io/struct.IoSliceMut.html#method.advance)
+- [`IoSliceMut::advance_slices`](https://doc.rust-lang.org/stable/std/io/struct.IoSliceMut.html#method.advance_slices)
+- [`PanicHookInfo`](https://doc.rust-lang.org/stable/std/panic/struct.PanicHookInfo.html)
+- [`PanicInfo::message`](https://doc.rust-lang.org/stable/core/panic/struct.PanicInfo.html#method.message)
+- [`PanicMessage`](https://doc.rust-lang.org/stable/core/panic/struct.PanicMessage.html)
+
+These APIs are now stable in const contexts:
+
+- [`char::from_u32_unchecked`](https://doc.rust-lang.org/stable/core/char/fn.from_u32_unchecked.html) (function)
+- [`char::from_u32_unchecked`](https://doc.rust-lang.org/stable/core/primitive.char.html#method.from_u32_unchecked) (method)
+- [`CStr::count_bytes`](https://doc.rust-lang.org/stable/core/ffi/c_str/struct.CStr.html#method.count_bytes)
+- [`CStr::from_ptr`](https://doc.rust-lang.org/stable/core/ffi/c_str/struct.CStr.html#method.from_ptr)
+
+### Compatibility notes
+
+#### Split panic hook and panic handler arguments
+
+We have renamed [`std::panic::PanicInfo`] to [`std::panic::PanicHookInfo`]. The old
+name will continue to work as an alias, but will result in a deprecation
+warning starting in Rust 1.82.0.
+
+`core::panic::PanicInfo` will remain unchanged, however, as this is now a
+*different type*.
+
+ The reason is that these types have different roles:
+`std::panic::PanicHookInfo` is the argument to the [panic hook](https://doc.rust-lang.org/stable/std/panic/fn.set_hook.html) in std
+context (where panics can have an arbitrary payload), while
+`core::panic::PanicInfo` is the argument to the
+[`#[panic_handler]`](https://doc.rust-lang.org/nomicon/panic-handler.html) in
+`#![no_std]` context (where panics always carry a formatted *message*). Separating
+these types allows us to add more useful methods to these types, such as
+[`std::panic::PanicHookInfo::payload_as_str()`]() and
+[`core::panic::PanicInfo::message()`](https://doc.rust-lang.org/stable/core/panic/struct.PanicInfo.html#method.message).
+
+[`std::panic::PanicInfo`]: https://doc.rust-lang.org/stable/std/panic/type.PanicInfo.html
+[`std::panic::PanicHookInfo`]: https://doc.rust-lang.org/stable/std/panic/type.PanicHookInfo.html
+
+#### Abort on uncaught panics in `extern "C"` functions
+
+This completes the transition started in [1.71](https://blog.rust-lang.org/2023/07/13/Rust-1.71.0.html#c-unwind-abi),
+which added dedicated `"C-unwind"` (amongst other `-unwind` variants) ABIs for
+when unwinding across the ABI boundary is expected. As of 1.81, the non-unwind
+ABIs (e.g., `"C"`) will now abort on uncaught unwinds, closing the longstanding soundess problem.
+
+Programs relying on unwinding should transition to using `-unwind` suffixed ABI
+variants.
+
+TODO: Check on status of https://github.com/rust-lang/rust/issues/123231
+
+#### WASI target naming changed
```

Comment:
Could we change this to:

```suggestion
#### WASI 0.1 target renamed
```

This way it's clear that this is not about the newer WASI 0.2 target, but just about the existing older WASI 0.1 targets. People tend to get confused if we don't qualify which WASI we refer to - especially since things are in a transitional state right now.